### PR TITLE
docs: Improve flat config ignores docs

### LINKED ISSUE #17822: Docs: (flat config) improve global ignores documentation
### Environment

Node version: v18.16.1
npm version: v9.5.0
Local ESLint version: v8.55.0 (Currently used)
Global ESLint version: Not found
Operating System: darwin 23.1.0

### What parser are you using?

Default (Espree)

### What did you do?

```
module.exports = [
  {
    ignores: ["dist"],
  },
  {
    files: ["src/**/*.js"],
    rules: {
      'no-console': 'error',
    }
  },
]
```

### What did you expect to happen?

Given the folder structure:
```
eslint.config.js
src/not-ignored.js
src/dist/not-ignored.js
```

I would expect that running `eslint src` would lint `src/not-ignored.js` and would skip `src/dist/not-ignored.js`.

### What actually happened?

Both files do get linted

```
$ ESLINT_USE_FLAT_CONFIG=true node_modules/.bin/eslint src

/Users/bjz/temp/eslint-flat-config-bugs/not-ignored-bug/src/dist/ignored.js
  1:1  error  Unexpected console statement  no-console

/Users/bjz/temp/eslint-flat-config-bugs/not-ignored-bug/src/not-ignored.js
  1:1  error  Unexpected console statement  no-console

✖ 2 problems (2 errors, 0 warnings)
```

### Link to Minimal Reproducible Example

https://github.com/bradzacher/eslint-flat-config-bugs/tree/main/not-ignored-bug

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

probably relevant DEBUG=* output

```
  @hwc/config-array Anonymous universal config found for /Users/bjz/temp/eslint-flat-config-bugs/not-ignored-bug/src/not-ignored.js +0ms
  @hwc/config-array Skipped config found for /Users/bjz/temp/eslint-flat-config-bugs/not-ignored-bug/src/not-ignored.js (based on ignores: **/node_modules/,.git/) +0ms
  @hwc/config-array Matching config found for /Users/bjz/temp/eslint-flat-config-bugs/not-ignored-bug/src/not-ignored.js +0ms
  @hwc/config-array Skipped config found for /Users/bjz/temp/eslint-flat-config-bugs/not-ignored-bug/src/not-ignored.js (based on ignores: dist) +0ms
  @hwc/config-array Matching config found for /Users/bjz/temp/eslint-flat-config-bugs/not-ignored-bug/src/not-ignored.js +1ms
  @hwc/config-array Anonymous universal config found for /Users/bjz/temp/eslint-flat-config-bugs/not-ignored-bug/src/not-ignored.js +0ms
  eslint:rules Loading rule 'no-console' (remaining=290) +0ms
  @hwc/config-array Anonymous universal config found for /Users/bjz/temp/eslint-flat-config-bugs/not-ignored-bug/src/dist/ignored.js +5ms
  @hwc/config-array Skipped config found for /Users/bjz/temp/eslint-flat-config-bugs/not-ignored-bug/src/dist/ignored.js (based on ignores: **/node_modules/,.git/) +0ms
  @hwc/config-array Matching config found for /Users/bjz/temp/eslint-flat-config-bugs/not-ignored-bug/src/dist/ignored.js +0ms
  @hwc/config-array Skipped config found for /Users/bjz/temp/eslint-flat-config-bugs/not-ignored-bug/src/dist/ignored.js (based on ignores: dist) +0ms
  @hwc/config-array Matching config found for /Users/bjz/temp/eslint-flat-config-bugs/not-ignored-bug/src/dist/ignored.js +0ms
  @hwc/config-array Anonymous universal config found for /Users/bjz/temp/eslint-flat-config-bugs/not-ignored-bug/src/dist/ignored.js +0ms
```

It looks like the `src/dist/ignored.js` file gets ignored then unignored by the `files` glob.

This is contrary to what I'd expect [based on the docs that state that](https://eslint.org/docs/latest/use/configure/configuration-files-new#globally-ignoring-files-with-ignores)

> If ignores is used without any other keys in the configuration object, then the patterns act as global ignores.

I wouldn't expect that I can implicilty unignore a file by simply adding it to a `files` - I'd expect that I'd need to explicitly unignore it eg `ignores: ['!src/dist/ignored.js']`

### PR DESCRIPTION
**Pull request by nzakas**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Adds more specifics to the documentation around ignoring directories and how patterns work in both the ignoring files flat config docs and the config migration docs.

fixes #17822

### CODE REVIEW

File: docs/src/use/configure/migration-guide.md [hunk 1]
```diff
@@ -545,7 +545,11 @@ export default [
 ];
 ```
 
-Also, with flat config, dotfiles (e.g. `.dotfile.js`) are no longer ignored by default. If you want to ignore dotfiles, add files ignore pattern `"**/.*"`.
+In `.eslintignore`, `temp.js` ignores all files named `temp.js`, whereas in flat config, you need to specify this as `**/temp.js`. The pattern `temp.js` in flat config only ignores a file named `temp.js` in the same directory as the configuration file.
+
+::: important
+In flat config , dotfiles (e.g. `.dotfile.js`) are no longer ignored by default. If you want to ignore dotfiles, add an ignore pattern of `"**/.*"`.
+:::
```
Comment (ecarrizo2) [inline docs/src/use/configure/migration-guide.md:552]: Not sure if the ::: adds too much value in this file (not very familiar with documentation practices of the project)

Below is a suggestion that looks clearer when previewing the MD at least for me.

```suggestion
*important*: When using flat config, dotfiles (e.g: `.dotfile.js`, `.env`) are not longer ignored by default. To ignore dotfiles use an ignore pattern of `"**/*`
```

Alternatively, it seems that your intention was to have a breakline from "important" from the text that clarify what is important, if that is the case you might need to fix the markup.

Review how this is looks now:
Screenshot 2024-01-19 at 13 51 03

Then perhaps this is the intention.
```
::: important

In flat config , dotfiles (e.g. `.dotfile.js`) are no longer ignored by default. If you want to ignore dotfiles, add an ignore pattern of `"**/.*"`.

:::

Comment (mdjermanovic) [reply docs/src/use/configure/migration-guide.md:552]: > Not sure if the ::: adds too much value in this file (not very familiar with documentation practices of the project)
> 
> Below is a suggestion that looks clearer when previewing the MD at least for me.

MD files are transformed into html for the eslint.org website. You can check how this looks like on the website here:

https://deploy-preview-17997--docs-eslint.netlify.app/use/configure/migration-guide

image

Comment (ecarrizo2) [reply docs/src/use/configure/migration-guide.md:552]: Oh I see thanks for the feedback

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**